assets/src/scripts/__tests__/components/ManagementForm.spec.tsx

```diff
@@ -0,0 +1,182 @@
+    const user = userEvent.setup();
+    render(<ManagementForm counts={defaultCounts} />);
+
+    const discardButton = screen.getByRole("button", {
+      name: "Discard",
+    });
+    await user.click(discardButton);
+
+    expect(
+      screen.getByText("Are you sure you want to discard this draft?"),
+    ).toBeVisible();
+    expect(screen.getByText("Discard draft")).toBeVisible();
+    expect(screen.getByText("Continue editing")).toBeVisible();
+  });
+
+  it("closes discard modal when clicking continue editing", async () => {
+    const user = userEvent.setup();
+    render(<ManagementForm counts={defaultCounts} />);
+
+    const discardButton = screen.getByRole("button", {
+      name: "Discard",
+    });
+    await user.click(discardButton);
+
+    const continueEditingButton = screen.getByRole("button", {
+      name: "Continue editing",
+    });
+    await user.click(continueEditingButton);
+
+    expect(
+      screen.queryByText("Are you sure you want to discard this draft?"),
+    ).not.toBeInTheDocument();
+  });
+
+  it("includes CSRF token in all forms", async () => {
+    const user = userEvent.setup();
+    render(<ManagementForm counts={defaultCounts} />);
+
+    let csrfInputs = screen.getAllByDisplayValue("test-csrf-token");
+    expect(csrfInputs).toHaveLength(1); // One in main form
+    csrfInputs.forEach((input) => {
+      expect(input).toHaveAttribute("name", "csrfmiddlewaretoken");
+      expect(input).toHaveAttribute("type", "hidden");
+    });
+
+    const discardButton = screen.getByRole("button", {
+      name: "Discard",
+    });
+    await user.click(discardButton);
+
+    csrfInputs = screen.getAllByDisplayValue("test-csrf-token");
+    expect(csrfInputs).toHaveLength(2); // One in main form, one in modal form
+    csrfInputs.forEach((input) => {
+      expect(input).toHaveAttribute("name", "csrfmiddlewaretoken");
+      expect(input).toHaveAttribute("type", "hidden");
+    });
+  });
+
+  it("includes correct action values in forms", async () => {
+    const user = userEvent.setup();
+    const countsWithAllResolved = {
+      ...defaultCounts,
+      "+": 1,
+      "-": 1,
+      total: 2,
+    };
+    render(<ManagementForm counts={countsWithAllResolved} />);
+
+    // Main form actions
+    const saveForReviewButton = screen.getByRole("button", {
+      name: "Save for review",
+    });
+    const saveDraftButton = screen.getByRole("button", {
+      name: "Save draft",
+    });
+
+    expect(saveForReviewButton).toHaveAttribute("value", "save-for-review");
+    expect(saveDraftButton).toHaveAttribute("value", "save-draft");
+
+    // Modal form action
+    const discardButton = screen.getByRole("button", {
+      name: "Discard",
+    });
+    await user.click(discardButton);
+
+    const discardDraftButton = screen.getByText("Discard draft");
+    expect(discardDraftButton.closest("form")).toContainElement(
+      screen.getByDisplayValue("discard"),
+    );
```

Comment:
Am I right that this is basically testing that: The form containing the "Discard draft" button, also contains an element with a `value` of `discard`? If so, what's the reasoning for doing it this way, rather than by checking the button with the `toHaveAttribute` method as with the other buttons in this test?

Reply:
That was a mistake, this was my initial way of testing the form, which I then updated to use the `toHaveAttribue`, but missed this last one. I'll fix it now.